### PR TITLE
Show the deprecated functions in the API docs

### PR DESCRIPTION
**Pull request by hawkowl**

### CODE REVIEW

**Comment by hawkowl** [comment]
This has been running on Twisted prod, so I'll just merge this...

File: pydoctor/twistedmodel.py [hunk 1]
```diff
@@ -25,12 +26,66 @@ def visitCallFunc_twisted_python_util_moduleMovedForSplit(self, funcName, node):
...
+            if isinstance(a, ast.CallFunc):
+                decorator = a.asList()
+                fn = cls.expandName(decorator[0].name)
+                if fn == "twisted.python.deprecate.deprecated":
+                    cls._deprecated_info = deprecatedToUsefulText(cls.name, decorator)
+
+
+
+def versionToUsefulText(version):
+    from twisted.python.versions import Version
```
Comment (mwhudson) [inline pydoctor/twistedmodel.py]: Why the function level import?

File: pydoctor/twistedmodel.py [hunk 2]
```diff
@@ -25,12 +26,66 @@ def visitCallFunc_twisted_python_util_moduleMovedForSplit(self, funcName, node):
...
+
+    return Version(*[x.value for x in version.asList()[1:] if x])
+
+
+def deprecatedToUsefulText(name, deprecated):
+
+    from twisted.python.deprecate import _getDeprecationWarningString
```
Comment (mwhudson) [inline pydoctor/twistedmodel.py]: Ditto

File: pydoctor/templates/nameIndex.html [hunk 3]
```diff
@@ -20,6 +20,13 @@
       </div>
     </nav>
 
+    <div style="display: none" id="current-docs-container" class="container">
+      <div class="col-sm-12">
+        <a id="current-docs-link">
+          Go to the latest version of this document.
+        </a>
+      </div>
+    </div>
```
Comment (mwhudson) [inline pydoctor/templates/nameIndex.html]: Not relevant to this PR.

File: pydoctor/templates/nameIndex.html [hunk 4]
```diff
@@ -39,6 +46,7 @@ <h2 t:render="letter" />
       </t:transparent>
 
     </div>
+
```
Comment (mwhudson) [inline pydoctor/templates/nameIndex.html]: Cleaner to remove this?

File: pydoctor/templatewriter/pages/functionchild.py [hunk 5]
```diff
@@ -31,10 +33,19 @@ def shortFunctionAnchor(self, request, tag):
...
         if self.ob.decorators:
-            decorators = [ast_pp.pp(dec) for dec in self.ob.decorators]
-        else:
-            decorators = []
+            for dec in self.ob.decorators:
+                if isinstance(dec.asList()[0], ast.Name):
+                    fn = self.ob.expandName(dec.asList()[0].name)
+                    # We don't want to show the deprecated decorator, it shows up
+                    # as an infobox
+                    print(fn)
```
Comment (mwhudson) [inline pydoctor/templatewriter/pages/functionchild.py]: Should probably take this out ;-)

File: pydoctor/templatewriter/pages/__init__.py [hunk 6]
```diff
@@ -88,12 +88,19 @@ def part(self):
         else:
             return []
 
-    def project(self):
+    def projectURL(self):
```
Comment (mwhudson) [inline pydoctor/templatewriter/pages/__init__.py]: I don't understand how/if the projectURL changes are related, maybe they're not?

File: pydoctor/templatewriter/pages/functionchild.py [hunk 7]
```diff
@@ -31,10 +33,19 @@ def shortFunctionAnchor(self, request, tag):
...
+                    if fn == "twisted.python.deprecate.deprecated":
```
Comment (mwhudson) [inline pydoctor/templatewriter/pages/functionchild.py]: Kinda an abstraction leak to have this here. But well, who really cares :-)

**Comment by mwhudson** [comment]
Hey, I made some comments, the core seems OK, there are some spurious changes in this PR though.

**Pull request merged**